File: Human_written_code/Code_with_comments/CPP/Q25.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    string s, t;
    cin >> s;
    bool isSymmetric = true;

    // Check if all characters are symmetric and if it's a palindrome
    for (char c : s) {
        if (c != 'A' && c != 'H' && c != 'I' && c != 'M' && c != 'O' && c != 'T' && c != 'U' && c != 'V' && c != 'W' && c != 'X' && c != 'Y') {
            isSymmetric = false;
            break;
        }
    }

    if (isSymmetric) {
        t = s;
        reverse(t.begin(), t.end());
        isSymmetric = (s == t);
    }

    cout << (isSymmetric ? "YES" : "NO") << endl;
    return 0;
}

File: Human_written_code/Code_with_comments/Python/Q25.py
def main():
    # Read input string and remove any leading or trailing whitespace
    s = input().strip()
    n = len(s)  # Get the length of the string
    flag = False  # Initialize a flag to indicate if the string is symmetrical
    check = True  # A flag to check if all characters are allowed symmetrical letters
    
    # Check if all characters are in the set of allowed symmetrical letters
    for char in s:
        if char not in {'A', 'H', 'I', 'M', 'O', 'T', 'U', 'V', 'W', 'X', 'Y'}:
            check = False  # Set check to False if any character is not symmetrical
            break  # Exit the loop if an invalid character is found

    if check:
        # Reverse the string and check if it matches the original
        t = s[::-1]
        flag = (s == t)  # Set flag to True if the string is the same as its reverse
    
    # Print "YES" if the string is symmetrical and contains only allowed characters, otherwise "NO"
    if flag:
        print("YES")
    else:
        print("NO")

# Entry point of the script
if __name__ == "__main__":
    main()
